controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1138,17 +1168,18 @@ private void checkState(State currState, State expectedState) {
                                                                     final Set<Integer> createdSegments,
                                                                     final int activeEpoch,
                                                                     final int newEpoch) {
-        return getHistoryTableFromStore()
+        return getHistoryIndexFromStore()
+                .thenCompose(historyIndex -> getHistoryTableFromStore()
                         .thenCompose(historyTable -> {
-                            final HistoryRecord lastRecord = HistoryRecord.readLatestRecord(historyTable.getData(),
+                            final HistoryRecord lastRecord = HistoryRecord.readLatestRecord(historyIndex.getData(), historyTable.getData(),
                                     false).get();
 
                             // idempotent check
                             if (lastRecord.getEpoch() > activeEpoch) {
-                                boolean idempotent = lastRecord.isPartial() &&
-                                        lastRecord.getSegments().containsAll(createdSegments) && lastRecord.getEpoch() == newEpoch;
+                                boolean idempotent = lastRecord.isPartial() && lastRecord.getSegments().containsAll(createdSegments);
```

Comment:
This term has been removed `lastRecord.getEpoch() == newEpoch`, why?

Reply:
it is already captured in the if condition `lastRecord.getEpoch() > activeEpoch`. 
Let me try to recollect if there was any other specific reason for the same though.